// angular2-validator-api/app/app.component.ts
import {Component} from "@angular/core"
import {ValidateDirective} from "./validate.directive"
@Component({
    selector : "myApp",
    template : `<h1>'Valiate' Directive Demo</h1>
                <div class="well well-sm">
                    <div class="form-group">
                        <label>Email</label>
                        <input type='text' validate='email' class="form-control" />
                    </div>
                    <div class="form-group">
                        <label>Number</label>
                        <input type='text' validate='number' class="form-control" />
                    </div>
                    <div class="form-group">
                        <label>Url</label>
                        <input type='text' validate='url' class="form-control" />
                    </div>
                    <div class="form-group">
                        <label>Phone(US)</label>
                        <input type='text' validate='usphone' class="form-control" />
                    </div>
                    <div class="form-group">
                        <label>Zip(US)</label>
                        <input type='text' validate='uszip' class="form-control" />
                    </div>
                    <div class="form-group">
                        <label>Credit Card</label>
                        <input type='text' validate='credit-card' class="form-control" />
                    </div>
                </div>
                `,
    directives: [ValidateDirective],
    host: {
        "(validated)" : "updateValidationResult($event)"
    }
})
export class AppComponent {
    updateValidationResult(event){
        console.log(event);
    }
}
