// angular2-validator-api/app/validate.directive.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var validator_service_1 = require('./validator.service');
var ValidateDirective = (function () {
    function ValidateDirective(el, validatorService) {
        // Emits validated event when a validation occurs
        this.validated = new core_1.EventEmitter();
        this.validationStarted = false;
        this.validationTypes = [];
        var that = this;
        that.el = el;
        that.validatorService = validatorService;
        var validationTypes = el.nativeElement.getAttribute("validate");
        if (validationTypes != null && validationTypes !== '' && typeof validationTypes != 'undefined') {
            validationTypes = validationTypes.trim().split(" ");
            validationTypes.forEach(function (t) {
                that.validationTypes.push(t);
            });
        }
        console.log(that.validationTypes);
    }
    ValidateDirective.prototype.initValidate = function () {
        if (!this.validationStarted) {
            this.validationStarted = true;
            this.onChange();
        }
    };
    ValidateDirective.prototype.onChange = function () {
        console.log(this.validationStarted);
        if (this.validationStarted) {
            console.log("Validation Result for : " + this.el.nativeElement.value);
            var validationResult = this.validatorService.validate(this.validationTypes, this.el.nativeElement.value);
            if (validationResult['finalResult']) {
                this.addClass(this.el.nativeElement, "valid");
                this.removeClass(this.el.nativeElement, "invalid");
            }
            else {
                this.addClass(this.el.nativeElement, "invalid");
                this.removeClass(this.el.nativeElement, "valid");
            }
            this.validated.emit(validationResult);
            console.log(this.el.nativeElement.value);
        }
    };
    ValidateDirective.prototype.addClass = function (element, classname) {
        var cn = element.className;
        //test for existance
        if (cn.indexOf(classname) != -1) {
            return;
        }
        //add a space if the element already has class
        if (cn != '') {
            classname = ' ' + classname;
        }
        element.className = cn + classname;
    };
    ValidateDirective.prototype.removeClass = function (element, classname) {
        var cn = element.className;
        var rxp = new RegExp("\\s?\\b" + classname + "\\b", "g");
        cn = cn.replace(rxp, '');
        element.className = cn;
    };
    __decorate([
        core_1.Output(), 
        __metadata('design:type', core_1.EventEmitter)
    ], ValidateDirective.prototype, "validated", void 0);
    ValidateDirective = __decorate([
        core_1.Directive({
            selector: '[validate]',
            host: {
                "(keyup)": "onChange()",
                "(change)": "initValidate()"
            },
            providers: [validator_service_1.ValidatorService]
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef, validator_service_1.ValidatorService])
    ], ValidateDirective);
    return ValidateDirective;
}());
exports.ValidateDirective = ValidateDirective;
