// angular2-validator-api/app/validate.directive.ts
import { Directive, ElementRef, Output, EventEmitter } from '@angular/core';
import { ValidatorService } from './validator.service'

@Directive({
    selector: '[validate]',

    host: {
        "(keyup)": "onChange()",
        "(change)": "initValidate()"
    },
    providers: [ValidatorService]
})
export class ValidateDirective {

    // Emits validated event when a validation occurs
    @Output() validated: EventEmitter<any> = new EventEmitter<any>()

    private validatorService: ValidatorService;
    private validationStarted: boolean = false;
    private el: ElementRef;
    private validationTypes: string[] = [];

    constructor(el: ElementRef, validatorService: ValidatorService) {
        var that = this;
        that.el = el;
        that.validatorService = validatorService;
        var validationTypes = el.nativeElement.getAttribute("validate");
        if (validationTypes != null && validationTypes !== '' && typeof validationTypes != 'undefined') {
            validationTypes = validationTypes.trim().split(" ");
            validationTypes.forEach(function (t) {
                that.validationTypes.push(t);
            });
        }
        console.log(that.validationTypes);
    }

    private initValidate() {
        if (!this.validationStarted) {
            this.validationStarted = true;
            this.onChange();
        }
    }

    private onChange() {
        console.log(this.validationStarted);
        if (this.validationStarted) {
            console.log("Validation Result for : " + this.el.nativeElement.value);
            var validationResult = this.validatorService.validate(this.validationTypes, this.el.nativeElement.value);
            if (validationResult['finalResult']) {
                this.addClass(this.el.nativeElement, "valid");
                this.removeClass(this.el.nativeElement, "invalid")
            } else {
                this.addClass(this.el.nativeElement, "invalid");
                this.removeClass(this.el.nativeElement, "valid")
            }
            this.validated.emit(validationResult);
            console.log(this.el.nativeElement.value);
        }
    }

    addClass(element, classname) {
        var cn = element.className;
        //test for existance
        if (cn.indexOf(classname) != -1) {
            return;
        }
        //add a space if the element already has class
        if (cn != '') {
            classname = ' ' + classname;
        }
        element.className = cn + classname;
    }

    removeClass(element, classname) {
        var cn = element.className;
        var rxp = new RegExp("\\s?\\b" + classname + "\\b", "g");
        cn = cn.replace(rxp, '');
        element.className = cn;
    }
}
