// angular2-validator-api/lib/app.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var validate_directive_1 = require("./validate.directive");
var AppComponent = (function () {
    function AppComponent() {
    }
    AppComponent.prototype.updateValidationResult = function (event) {
        console.log(event);
    };
    AppComponent = __decorate([
        core_1.Component({
            selector: "myApp",
            template: "<h1>'Valiate' Directive Demo</h1>\n                <div class=\"well well-sm\">\n                    <div class=\"form-group\">\n                        <label>Email</label>\n                        <input type='text' validate='email' class=\"form-control\" />\n                    </div>\n                    <div class=\"form-group\">\n                        <label>Number</label>\n                        <input type='text' validate='number' class=\"form-control\" />\n                    </div>\n                    <div class=\"form-group\">\n                        <label>Url</label>\n                        <input type='text' validate='url' class=\"form-control\" />\n                    </div>\n                    <div class=\"form-group\">\n                        <label>Phone(US)</label>\n                        <input type='text' validate='usphone' class=\"form-control\" />\n                    </div>\n                    <div class=\"form-group\">\n                        <label>Zip(US)</label>\n                        <input type='text' validate='uszip' class=\"form-control\" />\n                    </div>\n                    <div class=\"form-group\">\n                        <label>Credit Card</label>\n                        <input type='text' validate='credit-card' class=\"form-control\" />\n                    </div>\n                </div>\n                ",
            directives: [validate_directive_1.ValidateDirective],
            host: {
                "(validated)": "updateValidationResult($event)"
            }
        }), 
        __metadata('design:paramtypes', [])
    ], AppComponent);
    return AppComponent;
}());
exports.AppComponent = AppComponent;
